File: Java/Banking-java-dnbc5/src/banking/Checking.java
package banking;

import java.util.*;

import banking.exceptions.*;

// constructed using composition
public class Checking {

	private Account account;
	private static int nextCheckNbr = 100;
	private static ArrayList<Check> checks = new ArrayList<Check>();
	
	public void WriteCheck(String payee, double amount) throws InsufficientFundsException, DepositWithdrawTransferAmountNotPositiveException {
		Check check = new Check(nextCheckNbr++, new Date(), amount, this.getId(), payee);
		checks.add(check);
		this.withdraw(amount);
	}
	
	public int getId() {
		return account.getId();
	}
	public double getBalance() {
		return account.getBalance();
	}
	public String getDescription() {
		return account.getDescription();
	}
	public void setDescription(String newDescription) {
		account.setDescription(newDescription);
	}
	public void deposit(double amount) throws DepositWithdrawTransferAmountNotPositiveException {
		account.deposit(amount);
	}
	public void withdraw(double amount) throws InsufficientFundsException, DepositWithdrawTransferAmountNotPositiveException {
		account.withdraw(amount);
	}
	public boolean transferTo(Account toAccount, double amount) throws DepositWithdrawTransferAmountNotPositiveException {
		return account.transferTo(toAccount, amount);
	}
	
	public Checking(String newDescription) {
		account = new Account(newDescription);
	}
	public Checking() {
		account = new Account();
	}
}

File: BankingSolution/BankingProject/Checking.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BankingProject {
    class Checking : IAccount {

        private Account account { get; set; }
        private int NextCheckNbr { get; set; } = 1;

        public int GetId() {
            return account.GetId();
        }
        public decimal GetBalance() {
            return account.GetBalance();
        }
        public string GetDescription() {
            return account.GetDescription();
        }
        public void SetDescription(string NewDescription) {
            account.SetDescription(NewDescription);
        }
        public void Deposit(decimal Amount) {
            account.Deposit(Amount);
        }
        public void Withdraw(decimal Amount) {
            account.Withdraw(Amount);
        }

        public string Print() {
            return account.Print() + $", NextCheckNbr={NextCheckNbr}";
        }

        public int GetNextCheckNbr() {
            return NextCheckNbr;
        }

        public void TransferTo(decimal Amount, Account account) {

        }

        public Checking(Account acct) {
            account = acct;
        }

        //public Checking(string NewDescription) {
        //    account = new Account(NewDescription);
        //}
        public Checking() : this(null) {

        }
    }
}
